### PR TITLE
Adding System.IO.Hashing

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by mthalman** [comment]
For tracking purposes, what is new package needed for?

**Review by mthalman — COMMENTED** [review]
Per the instructions from https://github.com/dotnet/source-build-reference-packages#adding-new-packages:
* eng/Build.props needs to be updated to reference the new package version
* Modifications to existing package files need to be reverted

**Comment by CyrusNajmabadi** [comment]
Roslyn wants to use System.IO.Hashing to use the xxhash128 type to do content hashing.  We currently use sha256, but would benefit greatly from a faster hash (xxhash is around 40x faster). We are not using this for cryptographic purposes. Thanks!

**Comment by CyrusNajmabadi** [comment]
@mthalman does thsi look good now?

**Comment by MichaelSimons** [comment]
The addition of a non-stable version raises alarms with me.  source-build-reference-packages should be stable released versions.  Is there a reason why roslyn doesn't utilize a dependency flow for this package?

**Comment by CyrusNajmabadi** [comment]
>  Is there a reason why roslyn doesn't utilize a dependency flow for this package?

I don't know what "utilize a dependendency flow for this package" means.  Can you clarify?  

> The addition of a non-stable version raises alarms with me. 

The stable release (7.0) does not contain xxhash128, which is the library we are trying to move to in roslyn.

**Comment by MichaelSimons** [comment]
> I don't know what "utilize a dependendency flow for this package" means. Can you clarify?

Declare the dependency in the versions.details.xml file adorned with the source-build metadata indicating it is required by source-build and optionally define a darc subscription to automatically update it.

[Example](https://github.com/dotnet/roslyn/blob/main/eng/Version.Details.xml#L21)

If this is not feasible/desirable, I would be amenable to accepting this PR as long as there is a tracking issue logged to update the dependency to the released 8.0.0 version once it GAs.

**Comment by CyrusNajmabadi** [comment]
@MichaelSimons For context: i really don't know what any of these systems are doing, or the various pros/cons/reasonings surrounding any of this :)  I basically don't have any way to make an informed choice on what is best to do here.

For example "and optionally define a darc subscription".  I don't know what darc is.  I don't know what a darc subscription is :)

Basically, we simply want to reference this functionality so we can use it in roslyn, and that was blocked with a link saying we needed to update this repo. :)

> If this is not feasible/desirable,

I cannot make any determinations of feasibility or desirability.  I lack total context on being able to weigh in on that :)

> I would be amenable to accepting this PR as long as there is a tracking issue logged to update the dependency to the released 8.0.0 version once it GAs.

My end goal is simply being able to use the xxhash128 type, in a fashion that allows us to consume it on roslyn, while also meeting the needs of this repo.  But i'll need major assistance in getting there as I basically have no understanding of what the paths are within this repo, and i definitely cannot make determinations about which of those paths is the appropriate one to take :)

**Comment by MichaelSimons** [comment]
Thinking about this I think the correct action is to accept this PR and log a follow up issue to upgrade the roslyn reference to the 8.0 GA version once it is released.  If the roslyn PR can wait, perhaps just waiting until Tuesday would be less work - your call but let us know so that we can merge this PR for you or if we should wait.

FYI to learn about dependency flow please see https://github.com/dotnet/arcade/blob/main/Documentation/DependencyFlowOnboarding.md.

**Comment by CyrusNajmabadi** [comment]
We can wait until tuesday.  That's no issue for us.  :)

File: eng/Build.props
Comment (CyrusNajmabadi) [inline eng/Build.props]:
```suggestion
    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.IO.Hashing.8.0.0.csproj" />
```

**Comment by CyrusNajmabadi** [comment]
@mthalman @MichaelSimons this is updated to System.IO.Hashing 8.0.0.  Could you ptal?  Roslyn would very much like this.  Thanks! <3 :)

**Review by MichaelSimons — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**
